### ISSUE TITLE
Is there any way to make this Widget implement a specified initial Tab index?

### ISSUE THREAD

**Issue by NAOXUAN4** [afonsocraposo/buttons_tabbar#68]
When using Flutter's built-in TabBar, `animateTo` can be used `initState()` to specify the initial Tab, but using the same operation in buttons_tabbar seems to have no effect. Although the Tab pointed to by the initialized index has been highlighted, the animation of buttons_tabbar is not initialized in the expected position. Only when sliding TabBarView or clicking on other Tabs, the animation is initialized to the correct index.

Can you tell me how to specify the initial tab and correctly initialize the animation，or add a function to initialize the bar animation with a specified index? I hope you can provide assistance.🥰

```Dart
@override
void initState() {
  super.initState();
  _tabController = TabController(
      length: 10,
      vsync: this,
      initialIndex: 4);

  _tabController.animateTo(
    4,
    duration: const Duration(milliseconds: 100),
    curve: Curves.easeInOut,
  );
}
```
### 1. actual situation when starting app
![1](https://github.com/user-attachments/assets/c168b4b6-ad90-4e32-ab50-f7b91857c76a)

### 2. The desired effect to be achieved（when starting app）
![2](https://github.com/user-attachments/assets/e4de6e71-43e5-4056-8007-98717af16d16)

**Comment by afonsocraposo** [comment]
Hey @NAOXUAN4 !
Can you clarify a bit better what's the issue that you're having and what you want to achieve?
Did you add the controller to the `ButtonsTabBar` component? I tried the following with success:
```dart
import 'package:buttons_tabbar/buttons_tabbar.dart';
import 'package:flutter/material.dart';

class Example extends StatefulWidget {
  const Example({super.key});

  @override
  _ExampleState createState() => _ExampleState();
}

class _ExampleState extends State<Example> with SingleTickerProviderStateMixin {
  late TabController _tabController;

  @override
  void initState() {
    super.initState();
    _tabController = TabController(
      length: 6,
      vsync: this,
      initialIndex: 4,
    );
  }

  @override
  Widget build(BuildContext context) {
    const typeList = ["foo", "bar", "baz", "qux", "quux", "quuz"];
    return Scaffold(
      body: SafeArea(
        child: Column(
          children: <Widget>[
            ButtonsTabBar(
              controller: _tabController,
              backgroundColor: Colors.red,
              buttonMargin: const EdgeInsets.only(right: 16),
              contentPadding:
                  const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
              tabs: typeList.map((e) {
                return Tab(
                  child: Text(
                    e,
                    style: const TextStyle(
                      fontWeight: FontWeight.w500,
                      fontSize: 14,
                    ),
                  ),
                );
              }).toList(),
            ),
            Expanded(
              child: TabBarView(
                controller: _tabController,
                children: typeList.map((e) {
                  return Center(
                    child: Text(e),
                  );
                }).toList(),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

**Comment by NAOXUAN4** [comment]
Thank you for your response! 🥰 Let me explain the issue I'm encountering more clearly: There is no problem when all tabs can be displayed on a single page. However, when a page cannot accommodate all tabs (as shown in the example below, where I modified the tab length in the code you provided to 120 or any length that prevents all tabs from being displayed on one page), if the tab that needs to be highlighted upon initialization is not within the visible page area, there will be no animation to bring it into view.
![q](https://github.com/user-attachments/assets/618f8047-fce4-49bf-9a00-42a160d8a050)

However, in the native TabBar provided by Flutter, you can manually add the `_tabController.animateTo(index);` method in `initState` to manually initialize the position of the Tab.
![171373108fde12b2d25cbc003b1f5e77](https://github.com/user-attachments/assets/7d53225d-18c1-4751-a874-b64450a5c4f5)
![124a4ba6635cd8b211e4e29b020a5885](https://github.com/user-attachments/assets/4b2198d0-bba3-48af-9f88-cd2ce23a2982)

However, when using `ButtonTabBar`, adding the same method（_tabController.animateTo(index);） in `initState` does not correctly initialize the tab position.

**Comment by bishal-rumba007** [comment]
any updates on this?

**Comment by afonsocraposo** [comment]
Hey @NAOXUAN4 and @bishal-rumba007! I'm sorry for the delay. These last weeks were very chaotic: work, advent of code, my father passing away, Christmas, etc.
Thank you for pushing this.
I fixed this issue in the latest version `1.3.15`. Give it a try and let me know. 
Feel free to reopen the issue if needed. Thank you!
